### DISCUSSION TITLE
IAM Role for S3 Compatible Storage

### DISCUSSION THREAD

**Discussion by joshuahigginson1** [logto-io/logto#5751] [Q&A]
Hi Team,

The guide for configuring S3-Enabled storage relies on passing the logto CLI application and Access Key ID and a Secret Access Key. This secret access key is then stored within our database.

We are hosting LogTo on ECS Fargate, and was wondering if there was the option to use the IAM Task Role as a method of authentication for our S3 bucket (Think IAM InstanceProfile in AWS EC2, but for ECS containers, or IRSA for EKS pods).

Many thanks,

Josh

**Comment by darcyYe** [comment]
@wangsijie would you like to take a look on this since you may familiar on AWS, thanks!

**Comment by wangsijie** [comment]
Thanks for the suggestion, I'll add this to the roadmap, you can also make a contribution by making a pull request.
